### DISCUSSION TITLE
Why is `etc/zfs/Makefile.am` missing in the 2.2.0-rc3 tag?

### DISCUSSION THREAD

**Discussion by colemickens** [openzfs/zfs#15207] [General]
What's going on with the branch management for ZFS?

I'm trying to figure out why 1.1.3-staging has `etc/zfs/Makefile.am`, but `2.2.0-rc3` does not, and I can't seem to find any history removing it? Is the 1.1.x lineage carrying something that adds that file that wasn't ported over to `2.2.0-rc3`? Is there some change to the build system for 2.2.x that I can read up on to further understand, and then spend more of my limited life on autotools stuff?

Thanks in advance. And thanks for the best FS, ~ever~ so far. :)

**Comment by rincebrain** [comment]
1.1.x is not a release OpenZFS has ever had.

If you mean 2.1.x, see 48f4379974d5b28bba9c22fcf3bab6ceb2844ff7

**Comment by colemickens** [comment]
Sorry, I definitely crossed my wires when posting, I did mean the 2.1.x series.

For example:
* https://github.com/openzfs/zfs/tree/zfs-2.2.0-rc3/etc/zfs - No `Makefile.am`
* https://github.com/openzfs/zfs/tree/zfs-2.1.12/etc/zfs - Contains `Makefile.am`

and the commit removing the Makefile in the 2.2 series is from... 18 months ago.

**Reply by rincebrain** [reply]
15 and 18 are different numbers of months.

2.1 branched from master in March 2021, I believe. Usually a release branch branches from master at some point and starts creating RCs, with backported fixes from master as needed, until it becomes a stable release, and then more backports ensue as the life of that branch continues, until it's eventually no longer updated sometime after the next release is branched and has a stable release.

**Comment by colemickens** [comment]
I guess the 2.2.x series has been in development for a while. Life is too short to waste any more of it on autotools, I guess I'll just pin this generation that I have that works and boots with my upgraded pool and wait for someone else to sort this out.

thanks

**Reply by rincebrain** [reply]
You never specified why you care about this specific file being absent in the first place.
